cmd/gitcloner/gogit/cloner_test.go

```diff
@@ -0,0 +1,139 @@
+package gogit
+
+import (
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/go-git/go-git/v5"
+	httpgit "github.com/go-git/go-git/v5/plumbing/transport/http"
+	gossh "github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/google/go-cmp/cmp"
+	"github.com/rancher/gitjob/cmd/gitcloner/cmd"
+)
+
+func TestCloneRepo(t *testing.T) {
+	const (
+		passwordFile             = "passFile"
+		passwordFileContent      = "1234"
+		sshPrivateKeyFile        = "sshFile"
+		sshPrivateKeyFileContent = `-----BEGIN RSA PRIVATE KEY-----
+MIICXQIBAAKBgQC1ZuFGlFeAFqeS6p04QsliOXG3NH1/lQC4UMXdQ0F73ciYBPKq
+iQZcoyOu8a2Hsi5HvxDqR1rreTAkJ37C3ErrmKcE1CUJwxBVqkgE17Fzw63QBu0X
+0OVtaUarG8Pd9HuKbXPK8HXFTEh6F5hoqmzCmG7cRHmagBeh1SqZm1awzQIDAQAB
+AoGAChHZ84cMjGm1h6xKafMbJr61l0vso4Zr8c9aDHxNSEj5d6beqaTNm5rawj1c
+Oqojc4whrj+jxmqFx5wBp2N/LRi7GhpPco4wy8gg2t/OjmcR+jTRJgT1x1Co9W58
+U+O5c001YFTNoa1UUUBweqye/sX/k5GBCUt0V2G839Cn+8ECQQD2K2eZcyUeeBHT
+/YhGAq++mmfVEkzMY7U+G59oeF038zXX+wtMwoKmC9/LHwVPWpnzL/oMu3zZqv4a
+jzCOAdZpAkEAvKVas8KUctHUBvDoU6hq9bVyIZMZZnlBfysuFEeJLU8efp/n4KRO
+93EyhcXe2FmOC/VzGbkiQobmAqVvIwTixQJBAIKYZE20GG0hpdOhHTqHElU79PnE
+y5ljDDP204rI0Ctui5IZTNVcG5ObmQ5ZVqfSmPm66hz3GjUf0c6lSE0ODIECQHB0
+silO6We5JggtPJICaCCpVawmIJIx3pWMjB+StXfJHoilknkb+ecQF+ofFsUqPb9r
+Rn4jGwVFnYAeVq4tj3ECQQCyeMeCprz5AQ8HSd16Asd3zhv7N7olpb4XMIP6YZXy
+udiSlDctMM/X3ZM2JN5M1rtAJ2WR3ZQtmWbOjZAbG2Eq
+-----END RSA PRIVATE KEY-----`
+	)
+	var (
+		pathCalled      string
+		isBareCalled    bool
+		cloneOptsCalled *git.CloneOptions
+	)
+
+	sshAuth, _ := gossh.NewPublicKeys("git", []byte(sshPrivateKeyFileContent), "")
+	sshKeyComparer := cmp.Comparer(func(x, y gossh.PublicKeys) bool {
+		return x.User == y.User &&
+			x.Signer.PublicKey().Type() == y.Signer.PublicKey().Type() &&
+			cmp.Equal(x.Signer.PublicKey().Marshal(), y.Signer.PublicKey().Marshal())
+	})
+
+	plainClone = func(path string, isBare bool, o *git.CloneOptions) (*git.Repository, error) {
+		pathCalled = path
+		isBareCalled = isBare
+		cloneOptsCalled = o
+
+		return &git.Repository{}, nil
+	}
+	readFile = func(name string) ([]byte, error) {
+		if name == passwordFile {
+			return []byte(passwordFileContent), nil
+		}
+		if name == sshPrivateKeyFile {
+			return []byte(sshPrivateKeyFileContent), nil
+		}
+		return nil, errors.New("file not found")
+	}
```

Comment:
Why replacing `readFile` instead of just creating a file with that contents and set their path in the Options?

Reply:
I think it's better to mock the readFile than actually create a file for unit testing. That way the test is more isolated, and we don't need to create files in the machine that runs the tests.